src/Pages/CardReader/CardReader.js

```diff
@@ -34,25 +116,108 @@ export default function CardReader() {
       }
     };
 
-    eventSource.onerror = () => {
-      setError('Error connecting to SSE');
+    eventSource.onerror = (error) => {
+      setLogs(
+        (currLogs) => [
+          '...crickets...',
+          ...currLogs
+        ]
+      );
+      setConnected(false);
     };
 
     return () => {
       eventSource.close();
     };
-
   }, []);
 
+  useEffect(() => {
+    if (connected) {
+      setConnectionStatusText('Listening for card reader activity...');
```

Comment:
```suggestion
      setConnectionStatusText('Listening for card reader activity...');
      return;
```
then you can remove the else block

Reply:
done